ARCHITECTURE.md

```diff
@@ -0,0 +1,158 @@
+# Istio Architecture
+
+This document describes Istio's end-to-end design, to help you understand what it does and how
+it does it.
+
+- [Problem statement](#problem-statement)
+- [High-level composition](#high-level-composition)
+- [Attributes](#attributes)
+- [Configuration](#configuration)
+- [The proxy](#the-proxy)
+- [The mixer](#the-mixer)
+  - [Adapters](#adapters)
+- [The manager](#the-manager)
+- [Istio and Kubernetes](#istio-and-kubernetes)
+
+## Problem statement
+
+As we move towards a world where cloud developers think in terms of microservices, rather than in terms of low level resources like VMs and firewalls, there 
+is a need for a fabric that can take care of common cross-cutting capabilities around service communication and management such as secure interconnect, load 
+balancing, service discovery, rate limiting, authorization and access control, monitoring, logging, etc. Having a common, open and uniform way to enable these
+capabilities across services, independent of environment, makes services thinner and portable, teams more agile and focused on core business logic, enables 
+sweeping changes through configuration, and makes it easy to monitor the entire service fabric uniformly. Such a fabric can help enterprises have a shared 
+horizontal layer which can be maintained by central ops teams and make service teams and code lighter and network agnostic.
+
+## High-level composition
+
+Istio consists of three major components:
+
+- **Proxy**. The Istio proxy is designed to mediate inbound and outbound traffic for all Istio-managed services. It is used to enforce
+access control and use policies and provides protocol conversion. The Istio proxy is based on the Envoy proxy with appropriate 
+extensions to fit within the Istio service mesh.
+
+- **Mixer**. The Istio mixer provides the foundation of the Istio service mesh design. It is the component responsible for
+evaluating Istio policies and abstracting details of the execution environment from the proxy, the manager, and Istio-managed services via
```

Comment:
The term 'execution environment' seems a little vague. Also not sure that we care too much about abstracting details from the manager. A slight variation ....

The Istio Mixer is the nexus of the Istio service mesh design. The proxy delegates policy decisions to the Mixer which in turn delegates to it's configured policy plugins. Similarly the Mixer receives metrics and logs from the proxy which it federates to downstream analysis tools.